File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        dict  = {}
        for i in range(0,len(nums)):
            ele = target - nums[i]
            if ele in dict:
                return [i, dict[ele]]
            else:
                dict[nums[i]] =  i       

File: 0001-two-sum/0001-two-sum.js
var twoSum = function(nums, target) {
    // Map elements to objects containing value and original index
    let indexedNums = nums.map((num, index) => ({ value: num, index: index }));
    
    // Fix: Sort numerically using a comparator function
    indexedNums.sort((a, b) => a.value - b.value);
    
    let i = 0;
    let j = indexedNums.length - 1;
    
    while (i < j) { // Changed to i < j since elements cannot pair with themselves
        let sum = indexedNums[i].value + indexedNums[j].value;
        
        if (sum === target) {
            return [indexedNums[i].index, indexedNums[j].index]; // Return original indices
        } else if (sum < target) {
            i++;
        } else {
            j--;
        }
    }
};